pkg/registry/core/serviceaccount/storage/token.go

```diff
@@ -90,7 +90,7 @@ func (r *TokenREST) Create(ctx genericapirequest.Context, name string, obj runti
 			return nil, errors.NewBadRequest(fmt.Sprintf("cannot bind token to object of type %s", gvk.String()))
 		}
 		if ref.UID != "" && uid != ref.UID {
-			return nil, errors.NewConflict(schema.GroupResource{Group: gvk.Group, Resource: gvk.Kind}, ref.Name, fmt.Errorf("the UID in the bound object reference (%s) does not match the UID in record (%s). The object might have been deleted and then recreated", ref.UID, uid))
```

Comment:
We are not sure the user are authoried to read pod.Spec.ServiceAccountName pod.UID or secert.UID.
So, we should not let users known this information, I think.

Reply:
I think this should still return a conflict, even if it doesn't echo back the current UID

Reply:
Done.